### PR TITLE
Invite-only mode with auto-activation for mozilla.com emails and vouched Mozillians

### PR DESCRIPTION
**Pull request by lmorchard**
TL;DR: This PR adds an invite-only mode to the site with exceptions for pre-defined qualifications. 

Broke this up into some more bite-sized commits that might be easier to review in order. 

Anyone can login with FxA and have an account created. But, if they don't match qualifications to be auto-activated, their account will be flagged inactive and they will see a special page for inactive accounts. 

Users who match the qualifications will be auto-activated and can use the site. Currently, those qualifications are either having an @mozilla.com email address or being a vouched member of mozillians.org.

If we later turn off invitation mode, all previously inactive accounts will become auto-activated on the next login attempt. If a user gains qualification to be activated, they will also be auto-activated on the next login attempt - e.g. If they've been vouched on mozillians.org since the initial signin attempt.

Users can be manually activated in the Django admin tool.

Next steps from this PR:
- # 213 - The inactive account page is terrible. It should be reworked to look like mozilla/idea-town-splash and have some actually decent messaging
- a way to invite users in bulk with email notifications. We can handle that in a future PR that adds such a tool to the Django admin. We'll need a background job queue (#163) to make that happen, too.
- a way add users to an Idea Town mailing list (#129). But, we can come up with addresses for that list eventually with a query on user profiles with `invite_pending=True`

### CODE REVIEW

File: docker-compose-base.yml [hunk 1]
```diff
@@ -33,3 +33,5 @@ server:
     - FXA_SECRET_KEY=a93644d71fa684da464fd2dba4d687c14cb21e1dd740af83895e70ede82f54cc
     - DATADOG_API_KEY=4f3b967bbf13c7769ac4fa89efda0fae
     - DATADOG_APP_KEY=ddd45a7b3a3cb90ff1baf20ae8cfab04d1937038
+    - ACCOUNT_INVITE_ONLY_MODE=True
+    - MOZILLIANS_API_KEY=ff33fa1255ce0ad59be17271093ab851c6357936
```
Comment (pdehaan) [inline docker-compose-base.yml:37]: Sorry, ignorant question here, but are all these KEYS kosher in GitHub? Part of me mildly panics when I see `x_SECRET_KEY` or thinking that somebody could use our `MOZILLIANS_API_KEY` for their apps (although not really an issue if the keys only work for certain pre-approved URLs).

Comment (lmorchard) [inline docker-compose-base.yml:37]: I'm thinking they're okay, because we'll use different credentials & keys for production. They're just here for consistent dev environments that work out of the box without requiring new devs to dig up all these things on their own. Any data that ends up submitted or used with these APIs is throwaway.

**Comment by lmorchard** [comment]
Rebased against master

**Comment by lmorchard** [comment]
Rebase against master again again
